test/compile.jl

```diff
@@ -159,16 +159,23 @@ try
           """)
 
     Base.compilecache("FooBar")
-    sleep(2)
     @test isfile(joinpath(dir, "FooBar.ji"))
 
+    sleep(2) # wait to make sure file modification date changes
```

Comment:
you might need a bit more than 2 seconds to ensure the *date* changes

(not serious)